### PR TITLE
Access hbase tables from hive metadata. And support pre-loded hbase tables.

### PR DESCRIPTION
**Pull request by ravipesala**
1. Accessing hive tables with HbaseStorageHandler. 
 If the hive table is created with HbaseStorageHandler like below.
```
CREATE EXTERNAL TABLE TestHiveTable(
  strcol STRING,
  longcol BIGINT,
  floatcol FLOAT,
  doublecol DOUBLE )
 stored by 'org.apache.hadoop.hive.hbase.HBaseStorageHandler'
 WITH SERDEPROPERTIES (
 "hbase.columns.mapping" = "cf1:hstrcol#b,cf2:hlongcol#b,cf2:hfloatcol#b,:key#b" )
 TBLPROPERTIES ("hbase.table.name" = "TestHbaseTable")
```
User can access this table directly in Astro and queries the data from Astro.
Limitations  : 
1.Key column should be single column, not composite columns.
2.The data in Hbase table should be loaded with Hbase Bytes utility.
3.The hive columns should be binary format.
4.Not yet supported the following datatypes structs,map,array,union,timestamp,date .
5.Only it supports the SparkSQL standard SQL.
6.It cannot access the hive UDF and UDAF.

2. Supporting pre-loaded Hbase table in Astro.
User can create the Astro table with pre-loaded hbase as follows.
```
CREATE TABLE TestPreLoadedTable(
  strcol STRING,
  longcol LONG,
  floatcol FLOAT,
  doublecol DOUBLE
)
USING org.apache.spark.sql.hbase.HBaseSource
OPTIONS(
  hbaseTableName "TestHbaseTable",
  keyCols "doublecol",
  colsMapping "longcol=cf2.hlongcol, floatcol=cf2.hfloatcol, strcol=cf1.hstrcol, intcol=cf2.hintcol",
  encodingFormat "hbasebinaryformat"
)
```
While creating the table, user can mention the ```encodingFormat``` as ```hbasebinaryformat```, so that Astro uses Hbase Bytes utility to read the data from Hbase.

Limitations : 
1.Key column should be single column, if user wants it composite then it should comply the Astro keyformat(simply concatenating the values together, with a zero byte character used as a separator after a variable length type).

### CODE REVIEW

**Comment by ravipesala** [comment]
retest this please

**Comment by sathishvit** [comment]
retest this please

**Comment by ravipesala** [comment]
retest this please

**Comment by ravipesala** [comment]
retest this please

**Comment by ravipesala** [comment]
retest this please

**Comment by ravipesala** [comment]
Can any body review this PR?

File: src/main/scala/org/apache/spark/sql/hbase/util/DataTypeUtils.scala [hunk 1]
```diff
@@ -77,6 +83,8 @@ object DataTypeUtils extends Logging {
       case LongType => bu.toBytes(src.asInstanceOf[Long])
       case ShortType => bu.toBytes(src.asInstanceOf[Short])
       case StringType => bu.toBytes(src)
+      case DateType => bu.toBytes(src.asInstanceOf[Int])
```
Comment (Sephiroth-Lin) [inline src/main/scala/org/apache/spark/sql/hbase/util/DataTypeUtils.scala]: for a datetype value, like "2015-11-30" can convert ti int?

**Comment by ravipesala** [comment]
Please ignore Date and Timestamp support in this PR. I will raise another PR to support all data types

**Comment by Sephiroth-Lin** [comment]
I think good though it still have some limitations.

**Comment by ravipesala** [comment]
Thank you. Merged to spark1.5-alpha

**Pull request merged**